documentation/TechnicalDesignDocument.tex

```diff
@@ -44,7 +44,26 @@
 
     %------------------------------------------------------------------------------------------------------------------------------------------------------- Requirements
     \section{Requirements}
-    Functional and Non-functional requirements.
+    In this section the various functional and non-functional requirements the program has are listed.
+    \subsection{Functional Requirements:}
+    \begin{itemize}
+        \item The program should support loading different Marama shapes using the .mar files the Marama transpiler creates.
+        \item The program should let users add Marama objects to a world.
+        \item The program should let users see the Marama objects from any angle.
+        \item The program should allow users to attach Maramas to one another by their joints.
+        \item The program should check if attaching Maramas to one another is possible.
+        \item The program should support algebraic data structure Marama shapes.
+        \item The program should support polymorphic Marama shapes.
+        \item A game program should exist where the user can use the editor with predefined maps and limited Marama options.
+    \end{itemize}
+    \subsection{Non-Functional Requirements}
+    \begin{itemize}
+        \item The logic of the program should be written entirely in Scala.
+        \item The program should be made entirely open-source.
+        \item The program should be accessible on multiple platforms, mostly focusing on desktop and the Android platform.
+        \item A continuous integration system must be used to ensure that the program can properly build, even with multiple people working on it.
```

Comment:
It helps testing your codebase, automating your build process etc.

But it does not specifically help multiple people working together, that is more something provided by version control